modules/index-service/src/main/java/org/opencastproject/index/service/util/JSONUtils.java

```diff
@@ -198,6 +198,66 @@ public static JValue filtersToJSON(ResourceListQuery query, ListProvidersService
     return obj(filtersJSON);
   }
 
+  /**
+   * Generate JSON presentation of the given filters
+   *
+   * @param query
+   *          The {@link ResourceListQuery}
+   * @param listProvidersService
+   *          The {@link ListProvidersService} to get the possible values
+   * @param org
+   *          The {@link Organization}
+   * @param series
+   *          The Series with write access
+   * @return
+   * @throws ListProviderException
+   *           if the possible values can not be retrieved correctly from the list provider.
+   */
+  public static JValue filtersToJSONSeriesWriteAccess(ResourceListQuery query, ListProvidersService listProvidersService,
+          Organization org, Map<String, String> series) throws ListProviderException {
+
+    List<Field> filtersJSON = new ArrayList<Field>();
+    List<Field> fields = null;
+    List<ResourceListFilter<?>> filters = query.getAvailableFilters();
+
+    for (ResourceListFilter<?> f : filters) {
+      fields = new ArrayList<Field>();
+
+      fields.add(f("type", v(f.getSourceType().toString().toLowerCase())));
+      fields.add(f("label", v(f.getLabel())));
+
+      Option<String> listProviderName = f.getValuesListName();
+
+      if (listProviderName.isSome()) {
+        Map<String, String> values = null;
+        boolean translatable = false;
+
+        if (!listProvidersService.hasProvider(listProviderName.get())) {
```

Comment:
You could easily restructure this so that you do not have to invert the series map::

```diff
-        Map<String, String> values = null;
         boolean translatable = false;
 
-        if (!listProvidersService.hasProvider(listProviderName.get())) {
-          values = new HashMap<String, String>();
-        } else {
+        List<Field> valuesJSON = new ArrayList<>();
+        if (listProvidersService.hasProvider(listProviderName.get())) {
           if (listProviderName.get().equals("SERIES")) {
-            values = series;
+            for (Entry<String, String> entry : series.entrySet()) {
+              valuesJSON.add(f(entry.getValue(), v(entry.getKey(), Jsons.BLANK)));
+            }
           } else {
-            values = listProvidersService.getList(listProviderName.get(), query, org, false);
+            Map<String, String> values = listProvidersService.getList(listProviderName.get(), query, org, false);
+            for (Entry<String, String> entry : values.entrySet()) {
+              valuesJSON.add(f(entry.getKey(), v(entry.getValue(), Jsons.BLANK)));
+            }
           }
           translatable = listProvidersService.isTranslatable(listProviderName.get());
         }
 
-        List<Field> valuesJSON = new ArrayList<Field>();
-        for (Entry<String, String> entry : values.entrySet()) {
-          valuesJSON.add(f(entry.getKey(), v(entry.getValue(), Jsons.BLANK)));
-        }
```